packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -153,7 +153,14 @@ export default createRule({
      * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#parameters
      */
     function isTreatedAsZeroByArrayAt(value: unknown): boolean {
-      const asNumber = Number(value);
+      let asNumber: number;
+
+      try {
+        asNumber = Number(value);
+      } catch (e) {
```

Comment:
[Style] Nit: this will come into conflict with #8369.

```suggestion
      } catch {
```

Reply:
superseded by removal of try catch